### ISSUE TITLE
`Lint/LiteralAsCondition` make invalid autocorrections

### ISSUE THREAD

**Issue by m8o-d2a** [rubocop/rubocop#13918]
Maybe duplicate issue of #13917

But I'm reporting this because I'm having trouble with the hash.

--------

## Expected behavior

Do not change code.

## Actual behavior

```rb
aaa.rb:3:11: W: [Corrected] Lint/LiteralAsCondition: Literal { to_i: a.to_i } appeared as a condition.
  a: a && { to_i: a.to_i },
          ^^^^^^^^^^^^^^^^
```
```diff
{
-  a: a && { to_i: a.to_i },
+  a: a,
}
```

## Steps to reproduce the problem

```sh
$ cat aaa.rb
# frozen_string_literal: true

a = '1'
{
  a: a && { to_i: a.to_i },
}
$ rubocop aaa.rb
Inspecting 1 file
W

Offenses:

aaa.rb:5:11: W: [Correctable] Lint/LiteralAsCondition: Literal { to_i: a.to_i } appeared as a condition.
  a: a && { to_i: a.to_i },
          ^^^^^^^^^^^^^^^^
```

## RuboCop version

```
$ rubocop -V
1.73.0 (using Parser 3.3.7.1, rubocop-ast 1.38.1, analyzing as Ruby 3.4, running on ruby 3.4.1) [arm64-darwin24]
  - rubocop-performance 1.24.0
  - rubocop-rails 2.30.2
  - rubocop-rspec 3.5.0
  - rubocop-capybara 2.21.0
  - rubocop-factory_bot 2.26.1
  - rubocop-rspec_rails 2.30.0
```

**Comment by Zopolis4** [comment]
This should also be fixed by #19317.
